grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -0,0 +1,102 @@
+package org.pih.warehouse.inventory
+
+import grails.gorm.transactions.Transactional
+import grails.validation.ValidationException
+import org.grails.datastore.mapping.query.api.Criteria
+import org.hibernate.criterion.Order
+import org.hibernate.sql.JoinType
+import org.pih.warehouse.core.Location
+
+@Transactional
+class CycleCountService {
+
+    List<CycleCountCandidate> getCandidates(CycleCountCandidateFilterCommand command, String facilityId) {
+        if (command.hasErrors()) {
+            throw new ValidationException("Invalid params", command.errors)
+        }
+        Location facility = Location.read(facilityId)
```

Comment:
I assume "read" is like find but for read-only. Seems like we should be doing this more often :+1:

Reply:
For now none exception is thrown. Since we need the location id in the URL, it's a very rare case that the location would not exist. Even if it doesn't, the output would just be: 
```json
{
  data: [],
  totalCount: 0,
}
```
I'm fine with addressing this edge case later, if we are not comfortable with the empty result for non-existing location, which is barely to happen.